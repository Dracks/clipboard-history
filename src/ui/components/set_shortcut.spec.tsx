import { fireEvent, render, RenderResult } from '@testing-library/react'
import React from 'react'
import Subject from './set_shortcut'

describe('SetShortcut', ()=>{
    let query: RenderResult
    let onSave: jest.Mock

    beforeEach(()=>{
        onSave = jest.fn()
    })
    const mount = ()=>{
        query = render(<Subject title="Some Mock Title" save={onSave} value="peperoni" />)
    }

    it('Renders well', ()=>{
        mount()
        expect(query.container).toMatchSnapshot()
    })

    it('when click opens modal window', ()=>{
        mount()
        fireEvent.click(query.getByRole('textbox'))
        expect(query.container.parentElement).toMatchSnapshot("Showing the modal")
        fireEvent.click(query.getByText('Save and close'))
        expect(onSave).toBeCalledTimes(0)
    })

    it('when adding new shortcut it saves correctly', ()=>{
        mount()
        fireEvent.click(query.getByRole('textbox'))
        fireEvent.keyDown(window, {key: 'Alt'})
        fireEvent.keyDown(window, {key: 'A'})
        fireEvent.click(query.getByText('Save and close'))
        expect(onSave).toBeCalledTimes(1)
        expect(onSave).toBeCalledWith("Alt+A")
    })
})