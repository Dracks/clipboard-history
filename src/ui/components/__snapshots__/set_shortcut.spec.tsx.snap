// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetShortcut Renders well 1`] = `
<div>
  <div>
    <div
      class="euiFormControlLayout euiFormControlLayout--readOnly"
    >
      <div
        class="euiFormControlLayout__childrenWrapper"
      >
        <input
          class="euiFieldText"
          readonly=""
          type="text"
          value="peperoni"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`SetShortcut when click opens modal window: Showing the modal 1`] = `
<body
  class="euiBody-hasOverlayMask"
>
  <div>
    <div>
      <div
        class="euiFormControlLayout euiFormControlLayout--readOnly"
      >
        <div
          class="euiFormControlLayout__childrenWrapper"
        >
          <input
            class="euiFieldText"
            readonly=""
            type="text"
            value="peperoni"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="euiOverlayMask"
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="1"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        class="euiModal euiModal--maxWidth-default"
        tabindex="0"
      >
        <button
          aria-label="Closes this modal window"
          class="euiButtonIcon euiButtonIcon--text euiModal__closeIcon"
          type="button"
        >
          <div
            title="icon"
          >
            {"className":"euiButtonIcon__icon","type":"cross","size":"m","aria-hidden":"true"}
          </div>
        </button>
        <div
          class="euiModal__flex"
        >
          <div
            class="euiModalHeader"
          >
            <div
              class="euiModalHeader__title"
            >
              Some Mock Title
            </div>
          </div>
          <div
            class="euiModalBody"
          >
            <div
              class="euiModalBody__overflow"
            >
              <div>
                Press keys for the new shortcut. 
                <br />
                Selected: 
                0
                 
                <br />
                Not valid shortcut
              </div>
            </div>
          </div>
          <div
            class="euiModalFooter"
          >
            <button
              class="euiButton euiButton--primary"
              type="button"
            >
              <span
                class="euiButton__content"
              >
                <span
                  class="euiButton__text"
                >
                  Save and close
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;
